chef/cookbooks/crowbar/templates/default/crowbar-monasca-cleanups.sh.erb

```diff
@@ -0,0 +1,72 @@
+#!/bin/bash
+#
+# This script performs various Monasca cleanups required before
+# the Cloud 8 to Cloud 9 upgrade.
+#
+# The script should be run only on the monasca-server node.
+
+LOGFILE=/var/log/crowbar/node-upgrade.log
+UPGRADEDIR=/var/lib/crowbar/upgrade
+STATEFILE=${UPGRADEDIR}/crowbar-monasca-cleanups-ok
+
+mkdir -p "`dirname "$LOGFILE"`"
+exec >>"$LOGFILE" 2>&1
+
+log()
+{
+    set +x
+    echo "[$(date --iso-8601=ns)] [$$] $@"
+    set -x
+}
+
+install_new_persister()
+{
+    # Do nothing if java persister is not installed
+    rpm -q openstack-monasca-persister-java || return
+
+    zypper --non-interactive remove openstack-monasca-persister-java
+    local ret=$?
+
+    if [ $ret != 0 ]; then
+      log "Failed to remove Java based Monasca persister."
+      exit $ret
+    fi
+
+    zypper --non-interactive install openstack-monasca-persister
+    local ret=$?
+
+    if [ $ret != 0 ]; then
+      log "Failed to install Python based Monasca persister."
+      exit $ret
+    fi
+
+    # This is required because persister deinstallation will perform a userdel on
```

Comment:
I wonder if that isn't more like a bug.. can we just remove that code that deletes the user on postun? doesn't seem to make sense to me.

Reply:
We can, but we'd have to do that in the Cloud 8 package, which is the one doing the deleting.